cns/restserver/homeazmonitor.go

```diff
@@ -0,0 +1,168 @@
+package restserver
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/types"
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/Azure/azure-container-networking/nmagent"
+	"github.com/patrickmn/go-cache"
+	"github.com/pkg/errors"
+)
+
+const (
+	GetHomeAzAPIName = "GetHomeAz"
+	ContextTimeOut   = 2 * time.Second
+	homeAzCacheKey   = "HomeAz"
+)
+
+type HomeAzMonitor struct {
+	nmagentClient
+	values *cache.Cache
+	// channel used as signal to end of the goroutine for populating home az cache
+	closing chan struct{}
+	// channel used as signal to block the Start() of HomeAzMonitor until the first request to retrieve home az completes
+	block                    chan struct{}
+	cacheRefreshIntervalSecs time.Duration
+}
+
+// New creates a new HomeAzMonitor object
+func New(client nmagentClient, cacheRefreshIntervalSecs time.Duration) *HomeAzMonitor {
```

Comment:
should this be renamed to NewHomeAzMonitor? ( which will be inline with NewHTTPRestService )
with just new, the invocation looks like restserver.New() return HomeAzMonitor.